packages/autocertifier/src/CertificateCreator.ts

```diff
@@ -0,0 +1,111 @@
+import { Logger } from '@streamr/utils'
+import * as acme from 'acme-client'
+import fs from 'fs'
+import path from 'path'
+import { ChallengeManager } from './ChallengeManager'
+import { Certificate } from '@streamr/autocertifier-client'
+import os from 'os'
+
+const logger = new Logger(module)
+
+export class CertificateCreator {
+
+    private accountPrivateKey?: Buffer
+    private accountPrivateKeyPath: string
+    private readonly acmeDirectoryUrl: string
+    private readonly hmacKid: string
+    private readonly hmacKey: string
+    private readonly challengeManager: ChallengeManager
+
+    constructor(acmeDirectoryUrl: string,
+        hmacKid: string,
+        hmacKey: string,
+        privateKeyPath: string,
+        challengeManager: ChallengeManager
+    ) {
+        this.acmeDirectoryUrl = acmeDirectoryUrl
+        this.hmacKid = hmacKid
+        this.hmacKey = hmacKey
+        this.challengeManager = challengeManager
+
+        if (privateKeyPath.startsWith('~/')) {
+            this.accountPrivateKeyPath = privateKeyPath.replace('~', os.homedir())
+        } else {
+            this.accountPrivateKeyPath = privateKeyPath
+        }
+    }
+
+    public async createCertificate(fqdn: string): Promise<Certificate> {
+
+        logger.info(`Creating certificate for ${fqdn}`)
```

Comment:
Could adopt our loggin practices in both of the new packages: https://github.com/streamr-dev/network/blob/main/internal-docs/logging-practices.md

Reply:
Added to future improvements